types/src/system/auction.rs

```diff
@@ -125,6 +135,24 @@ impl BidsExt for Vec<BidKind> {
         }
     }
 
+    fn bridge(
+        &self,
+        public_key: &PublicKey,
+        new_public_key: &PublicKey,
+        era_id: &EraId,
+    ) -> Option<Bridge> {
+        if let BidKind::Bridge(bridge) = self.iter().find(|x| {
+            x.is_bridge()
+                && &x.validator_public_key() == public_key
+                && &x.new_validator_public_key().unwrap() == new_public_key
+                && &x.era_id().unwrap() == era_id
+        })? {
+            Some(*bridge.clone())
+        } else {
+            None
+        }
+    }
```

Comment:
If that's safe to unwrap, it would be good to document why. Alternatively, you could rewrite this into a `match` statement and a `find_map` to avoid unwraps since `new_validator_public_key` is only valid for a `Bridge` entry.

```suggestion
        self.iter().find_map(|x| {
            match x {
              BidKind::Bridge(bridge) if bridge.validator_public_key == public_key && bridge.new_validator_public_key == new_public_key && bridge.era_id == era_id { Some(bridge.clone()) }
              _ => None,
            }
        });
```